Mod2/Lessons/Week3/SQLAggregates.md

```diff
@@ -0,0 +1,158 @@
+> 4. 🌶️Spicy! How many artists are from 'New York'?
+> 5. 🌶️Spicy! How many different song lengths are represented?
+> 
+> Be ready to share out!
+
+### Sum
+
+`SUM()` adds all the values in a particular column and returns the total.
+
+> With a partner, use SUM() to answer the following questions:
+> 1. What is the total play_count of all songs?
+> 2. What is the total play_count of all songs with artist_id `2`?
+> 3. 🌶️ Spicy! What is the total play_count of all songs with a play_count less than 25?
+> 
+> Be ready to share out!
+
+
+### Average
+
+`AVERAGE()`  returns the mean (average) of all the numbers in a column.
+
+> With a partner, use AVERAGE() to answer the following questions:
+> 1. What is the average length of all songs?
+> 2. What is the average length of all songs with a play_count greater than 25?
+> 3. 🌶️ Spicy! What is the average play_count of songs for each artist?
+> 
+> Be ready to share out!
+
+## Grouping
+
+Grouping allows us to arrange data into sub-sections (groups) so that we can answer more specific questions.  In some of the exercises above, we needed to use a `WHERE` clause to drill down to the specific data we were looking for; in one question, we even needed to use multiple SQL statements!  Grouping allows us to answer questions about sub-sections of our data in one SQL statement.
+
+Using `GROUP BY` is a bit more complicated than only using Aggregate functions. As you build your query, you need to determine what to do (or not do) with every column in your table.
+
+** Instructor Note ** Instead of watching the video below, you could walk through an example in real-time.
+
+Watch [this short video](https://www.loom.com/share/4ce05601a3654b4c99ae2ecb40560fab) that walks through building a grouped aggregate statement on the `artists` table.  In the video, the narrator has one additional column to show that you _can_ group and include multiple aggregates if you want to or need to.  If you would like to try out a version of the SQL statement from the video, use the script below:
+
+```sql
+SELECT hometown, COUNT(name) FROM artists
+GROUP BY hometown;
+```
+
+<details><summary>Nice to Know: Aliasing</summary>
+    <p>
+    This is a nice-to-know, non-critical piece of information.
+    </p>
+    <p>
+    When you are using aggregates, it can be difficult sometimes to determine what information is showing up in your result set.  Try running the two SQL statements below:
+    </p>
+    <code>SELECT hometown, COUNT(name) FROM artists GROUP BY hometown;</code></br>
+    <code>SELECT hometown, COUNT(name) AS count_of_artists FROM artists GROUP BY hometown;</code>
+    </br>
+    </br>
+    <p>
+    Using `AS` allows you to rename the result column.  It <strong>does not</strong> change anything in the database table; it only appears in the results.
+    </p>
+</details>
+
+> With a partner, use GROUP BY and Aggregate functions, answer the following questions:
+> 1. For each artist, what is the average song length?
```

Comment:
As the instructor teaching this lesson I'd want to solve out the answers to these before class. If you already have the queries for these seems helpful to include them in the instructor notes, if not maybe the first instructor adds them.

This actually brings up a good question to of if it's helpful or not to include the solutions in drop downs for students to use in reviewing. Thoughts on if that's helpful or peeking is too tempting?

Reply:
I think we _either_ include a solution, or have instructors walk through it.  I'm more on the side of instructors walking through it; just because I want to avoid even the temptation of peeking.